tests/reg_tests/test_MPhysGeo.py

```diff
@@ -0,0 +1,566 @@
+        dvInfo = self.dvInfo
+
+        class ESPGroup(Group):
+            def setup(self):
+                self.add_subsystem("dvs", IndepVarComp(), promotes=["*"])
+                self.add_subsystem("geometry", OM_DVGEOCOMP(file=espBox, type="esp"))
+
+            def configure(self):
+                # get the DVGeo object out of the geometry component
+                DVGeo = self.geometry.nom_getDVGeo()
+
+                # add a point set on the surface
+                vertex1 = np.array([-2.0, -2.0, -2.0])
+                vertex2 = np.array([1.5, 1.5, 1.5])
+                left = np.array([-2.0, -1.1, -1.1])
+                right = np.array([1.5, -1.2, -0.1])
+                front = np.array([0.25, 1.5, 0.3])
+                back = np.array([1.2, -2.0, -0.3])
+                top = np.array([0.0, 0.1, 1.5])
+                bottom = np.array([-1.9, -1.1, -2.0])
+                self.initpts = np.vstack([vertex1, vertex2, left, right, front, back, top, bottom, left, right])
+
+                ptName = "mypts"
+                self.distglobal = self.geometry.nom_addPointSet(self.initpts.flatten(), ptName, cache_projections=False)
+                self.projPts = DVGeo.pointSets[ptName].proj_pts
+                DVGeo._updateModel()
+                DVGeo._updateProjectedPts()
+
+                for dv in dvInfo:
+                    if "upper" not in dv:
+                        dv["upper"] = None
+
+                    if "lower" not in dv:
+                        dv["lower"] = None
+
+                    if "scale" not in dv:
+                        dv["scale"] = None
+
+                    if "dh" not in dv:
+                        dv["dh"] = 0.001
+
+                    dvName = dv["name"]
+                    self.geometry.nom_addESPVariable(dvName, dh=dv["dh"])
+
+                    self.dvs.add_output(dvName)
+                    self.connect(dvName, f"geometry.{dvName}")
+
+                    self.add_design_var(dvName, upper=dv["upper"], lower=dv["lower"], scaler=dv["scale"])
+
+                self.add_constraint(f"geometry.{ptName}")
+
+        self.prob = Problem(model=ESPGroup())
+
+    def test_run_model(self):
+        self.prob.setup()
+        with self.assertRaises(Error):
+            try:
+                self.prob.model.geometry.nom_addESPVariable("cubew0")
+            except Error as e:
+                mes = e.message
+                raise e
+
+        self.assertEqual(mes, 'User specified design parameter name "cubew0" which was not found in the CSM file')
+
+        assert_near_equal(self.prob.model.distglobal, 0.0, 8)
+        assert_near_equal(np.linalg.norm(self.prob.model.initpts - self.prob.model.projPts), 0.0, 10)
+
+        self.prob.run_model()
+
+    def test_deriv_fwd(self):
+        self.prob.setup(mode="fwd")
+        self.prob.run_model()
+
+        totals = self.prob.check_totals(step=1e-7, out_stream=None)
+        assert_check_totals(totals)
```

Comment:
Can we add hard-coded tolerances for this check as well? Better to have them here than use the defaults from OM incase they change in the future.